downloads-generation/allele_sequences/filter_sequences.py

```diff
@@ -64,28 +49,46 @@ def run():
         for record in reader:
             total += 1
             name = record.description.split()[1]
-            normalized = normalize(name)
-            if not normalized and "," in record.description:
+            result = parse(name, raise_on_error=False)
+            if not result:
+                # TODO: are there other entries that require this?
+
                 # Try parsing uniprot-style sequence description
                 if "MOUSE MHC class I L-q alpha-chain" in record.description:
                     # Special case.
                     name = "H2-Lq"
-                else: 
-                    name = (
-                        record.description.split()[1].replace("-", "") +
-                        "-" +
-                        record.description.split(",")[-1].split()[0].replace("-",""))
-                normalized = normalize(name)
-            if not normalized:
-                print("Couldn't parse: ", name)
-                continue
-            if normalized in seen:
-                continue
-            if any(n in name for n in class_ii_names):
-                print("Dropping", name)
+                else:
+                    print("Unable to parse: %s" % name)
+                    continue
+            else:
+                if type(result) not in (Allele, AlleleWithoutGene, Gene):
```

Comment:
curious if you have an example for a `Gene` result? Should we really be including those or just skipping those sequences?

Reply:
Invariant genes sometimes don't have alleles. The one that comes to mind is "BoLA-DRA" but I've also seen it for non-canonical murine genes.